src/main/java/com/google/sps/data/GivrUser.java

```diff
@@ -104,22 +104,38 @@ public static void updateUserInDatastore(String identifyingProperty, String iden
     datastore.put(entity);
   }
 
+  public static GivrUser getUserByIdOrEmail(String userId, String userEmail) {
+    Entity entityRetrievedWithId = getUserFromDatastoreWithProperty("userId", userId);
+    Entity entityRetrievedWithEmail = getUserFromDatastoreWithProperty("userEmail", userEmail);
+
+    boolean isMaintainer = false;
+    boolean isLoggedIn = true;
+
+    if (entityRetrievedWithId != null) {
+      isMaintainer = (boolean) entityRetrievedWithId.getProperty("isMaintainer");
+    } else if (entityRetrievedWithEmail != null) {
+      isMaintainer = (boolean) entityRetrievedWithEmail.getProperty("isMaintainer");
+    }
+    
+    GivrUser user = new GivrUser(userId, isMaintainer, isLoggedIn, "" /* URL is not needed when User is logged in. */, userEmail);
+    return user;
+  }
+
   public static GivrUser getUserById(String userId) {
-    Entity entity = getUserFromDatastoreWithProperty("userId", userId);
+    Entity entityRetrievedWithId = getUserFromDatastoreWithProperty("userId", userId);
 
     boolean isMaintainer = false;
     boolean isLoggedIn = true;
     String userEmail = "";
-
-    if (entity != null) {
-      isMaintainer = (boolean) entity.getProperty("isMaintainer");
-      userEmail = (String) entity.getProperty("userEmail");
+    
+    if (entityRetrievedWithId != null) {
+      isMaintainer = (boolean) entityRetrievedWithId.getProperty("isMaintainer");
+      userEmail = (String) entityRetrievedWithId.getProperty("userEmail");
     }
 
     GivrUser user = new GivrUser(userId, isMaintainer, isLoggedIn, "" /* URL is not needed when User is logged in. */, userEmail);
     return user;
   }
-
   public static GivrUser getUserByEmail(String email) {
     // TODO: Support OAuth.
```

Comment:
Could this be simplified to

return getUserByIdOrEmail(/*userId=*/null, email);

?